### ISSUE TITLE
Post your screenshots / code here (PART 3)

### ISSUE THREAD

**Issue by ocornut** [ocornut/imgui#772]
**This is Part 3 out of 4. I am splitting issues to reduce loading times and locked the old ones.**
Screenshots Part 1 #123
Screenshots Part 2 #539
Screenshots Part 3 #772
Screenshots Part 4 #973

**You can post your screenshots here!**

**Comment by sylefeb** [comment]
Our online slicer for 3D printers (RepRaps, Ultimakers, Prusa, etc.), using the amazing ImGui! Try it here: http://shapeforge.loria.fr/slicecrafter/

![slicecrafter-imgui](https://cloud.githubusercontent.com/assets/13455188/17462457/ee7877c0-5cad-11e6-97e1-4a217104de43.jpg)

**Comment by eliasdaler** [comment]
I'm making a game called **Re:creation** (more info [here](http://en.sfml-dev.org/forums/index.php?topic=18062.0)).
Here are some in-game dev tools I've made with ImGui!

Level editor:
![leveleditor](http://i.imgur.com/iQibpSk.gif)

Animation editor:
![animationeditor](http://i.imgur.com/A44zLlb.gif)

---

I've also made [ImGui SFML binding](https://github.com/eliasdaler/imgui-sfml) and wrote two articles about ImGui ([Part 1](https://eliasdaler.github.io/using-imgui-with-sfml-pt1), [Part 2](https://eliasdaler.github.io/using-imgui-with-sfml-pt2))

**Comment by nem0** [comment]
Timeline Widget
![image](https://cloud.githubusercontent.com/assets/153526/17481789/18b71916-5d7f-11e6-9726-9e17dec1f3b4.png)
![timeline](https://cloud.githubusercontent.com/assets/153526/17481813/32f38f58-5d7f-11e6-9e28-e5b53ec316a2.gif)
Source code https://github.com/nem0/LumixEngine/blob/timeline_gui/external/imgui/imgui_user.inl#L814

**Comment by r-lyeh** [comment]
UI mockup for an asset library/manager I'm working on. Nothing functional as we speak :)
Built mostly by re-using code from omar (obviously), dougbinks, juliettef, itamago, simon geilfus, krys and flix01. 

Exciting times indeed!

![gif0](https://cloud.githubusercontent.com/assets/878187/17682734/e1cc697c-634d-11e6-918e-3d152bcd6a3a.gif)

**Comment by ratzlaff** [comment]
Nice!
@r-lyeh What do you use to make your gif animations?

**Comment by Flix01** [comment]
@r-lyeh Awesome main menu! And awesome icon font (or probably awesome usage of "Font Awesome" ?!?)
:)

**Comment by bkaradzic** [comment]
![animtool](https://cloud.githubusercontent.com/assets/814772/17825130/ee09ace8-661a-11e6-80b5-c315fa4eaaa6.png)

**Comment by damqui** [comment]
@bkaradzic nice ! is this anim tool somewhere on your github repos ?

**Comment by bkaradzic** [comment]
@damqui Nope, but all default settings for that UI are in bgfx repo.

**Comment by Flix01** [comment]
An unfinished (just the GUI ATM), very bad looking, Sudoku game for Dear ImGui:
![imguisudokugame](https://cloud.githubusercontent.com/assets/9608982/17868946/50223cb8-68b1-11e6-905f-d6c893767d14.png)

**Comment by phoenixcatdog** [comment]
Testing Imgui with native cocoa(no external dependencies) and opengl3(Working on Metal version).

![captura de pantalla 2016-08-24 a las 10 13 52](https://cloud.githubusercontent.com/assets/6180927/17923277/b5237482-69e3-11e6-8940-07f2441ae53e.png)

**Comment by edin-p** [comment]
Working on my 3D Model Editor/Game Engine, currently fixing my docking code and some bugs, for the most part it works.. imgui makes it very easy and beautiful...
![docking](https://cloud.githubusercontent.com/assets/12642134/18140511/288541a8-6fb6-11e6-8423-72b4c808016d.gif)

**Comment by thedmd** [comment]
This is sneek peak of node editor build on top of ImGui. It handle displaying and user interaction.
![node_preview_4](https://cloud.githubusercontent.com/assets/1197433/18221085/52f4c0d4-7179-11e6-9853-7f821089d6a5.gif)

ImGui can be bend to handle zooming while keeping crisp details.
![image](https://cloud.githubusercontent.com/assets/1197433/18220992/e2bb2e8a-7177-11e6-918b-f2a08411d24a.png)

**Comment by jarikomppa** [comment]
Squee! Please say the sources are/will be available?

**Comment by thedmd** [comment]
@jarikomppa Right now source isn't out there. I will think about making it open source.

**Comment by Flix01** [comment]
@thedmd I knew you were making awesome stuff! :)
> ImGui can be bend to handle zooming while keeping crisp details.

How?

**Comment by DrJedikiah** [comment]
![screenshot_2](https://cloud.githubusercontent.com/assets/12988451/18227831/6fbaa3b6-723a-11e6-9c26-b1ee0b848c5d.png)
I've edited ImGUI a lot, and i solve the AA issue thanks by the way it was about renderer. I changed my hook to Present[17] from EndScene. And this is dll file not exe. 

I made this tab system with buttons, i swapped the system of active state. If tab is active, button stays active too.

**Comment by thedmd** [comment]
@Flix01: You're correct. Branch 2016-09-fringe-scale is related to my solution. I'm scaling content of draw list vertex buffer. When zooming in finge is also magnified. To get rid of that I scale it by inverse of zoom factor. This is basic idea. I will post some code as an example when I get to my PC.

**Comment by Flix01** [comment]
@thedmd Thanks.

**Comment by thedmd** [comment]
@Flix01: In more detail. To achieve scaling choose some scale factor, set fringe to 1 / scale factor and draw as normal. I'm drawing on multiple channels to achieve layering but this isn't necessary. After drawing is done call `ImDrawList_TransformChannel_Inner` and voilà, you are done. Almost.

This works but some problems arise around it. Working with global coordinate system (GCS) when scaling is involved isn't best experience. Treating your child window as canvas which works in local coordinate system (LCS) is much easier. (0, 0) is in top left corner of child window. I advise to use matrices to represent transformation from one space to another from the start.

First thing is transforming `ImGuiIO::MousePos` and `ImGuiIO::MouseClickPos` to LCS. This will allow ImGui to handle input correctly.

Second. Push new clip rectangle which cover visible part of your canvas. When scaling down canvas is larger than actual window, input is clipped to window size in GCS. Pushing larger clip rectangle enable ImGui to handle larger areas. Whole scaeled area isn't visible yet but since `MousePos` is transfomed ImGui is actually processing input over whole area.

Third. Be ready to restore `ImGuiIO::MousePos` and `ImGuiIO::MouseClickPos` to original values and bring them back to LCS. It is necessary if you want show popups correctly.

Last thing to be done is to transform window draw list to make content actually visible. You can use `ImDrawList_TransformChannels` for that. If you want to write similar function by yourself make sure one vertex is transformed only once. If everything inside window is scalled, good. You can just transform vertices. If only some channels are drawn in LCS things are more tricky. Include code handle former.

If something is unclear or language I use is odd, please point me that out and ask questions.

```cpp
static void ImDrawList_TransformChannel_Inner(ImVector<ImDrawVert>& vtxBuffer, const ImVector<ImDrawIdx>& idxBuffer, const ImVector<ImDrawCmd>& cmdBuffer, const ImVec2& preOffset, const ImVec2& scale, const ImVec2& postOffset)
{
    auto idxRead = idxBuffer.Data;

    std::bitset<65536> indexMap;

    int minIndex    = 65536;
    int maxIndex    = 0;
    int indexOffset = 0;
    for (auto& cmd : cmdBuffer)
    {
        int idxCount = cmd.ElemCount;

        if (idxCount == 0) continue;

        for (int i = 0; i < idxCount; ++i)
        {
            int idx = idxRead[indexOffset + i];
            indexMap.set(idx);
            if (minIndex > idx) minIndex = idx;
            if (maxIndex < idx) maxIndex = idx;
        }

        indexOffset += idxCount;
    }

    ++maxIndex;
    for (int idx = minIndex; idx < maxIndex; ++idx)
    {
        if (!indexMap.test(idx))
            continue;

        auto& vtx = vtxBuffer.Data[idx];

        vtx.pos.x = (vtx.pos.x + preOffset.x) * scale.x + postOffset.x;
        vtx.pos.y = (vtx.pos.y + preOffset.y) * scale.y + postOffset.y;
    }
}

static void ImDrawList_TransformChannels(ImDrawList* drawList, int begin, int end, const ImVec2& preOffset, const ImVec2& scale, const ImVec2& postOffset)
{
    int lastCurrentChannel = drawList->_ChannelsCurrent;
    if (lastCurrentChannel != 0)
        drawList->ChannelsSetCurrent(0);

    auto& vtxBuffer = drawList->VtxBuffer;

    if (begin == 0 && begin != end)
    {
        ImDrawList_TransformChannel_Inner(vtxBuffer, drawList->IdxBuffer, drawList->CmdBuffer, preOffset, scale, postOffset);
        ++begin;
    }

    for (int channelIndex = begin; channelIndex < end; ++channelIndex)
    {
        auto& channel = drawList->_Channels[channelIndex];
        ImDrawList_TransformChannel_Inner(vtxBuffer, channel.IdxBuffer, channel.CmdBuffer, preOffset, scale, postOffset);
    }

    if (lastCurrentChannel != 0)
        drawList->ChannelsSetCurrent(lastCurrentChannel);
}
```

**Comment by Flix01** [comment]
@thedmd thanks again for your detailed explanation.
It's something I'll investigate further in the future... (I still have to find out how antialiasing works...).

P.S. I'd also like to manage the draw lists to see if sorting indices on a texture-id basis is worthy or not. 

@ocornut : sorry for having asked questions in the "Screenshot" section.

**Comment by thedmd** [comment]
This is from a weekend project [ascii-renderer](https://github.com/thedmd/ascii-render) where ImGui is used to control how rendering works. Only grayscale now.

![image](https://cloud.githubusercontent.com/assets/1197433/18396086/0aaa909e-76c2-11e6-99fe-27a11b6ed4a4.png)

And yes, you can render ImGui to ASCII buffer. : )
![image](https://cloud.githubusercontent.com/assets/1197433/18396091/10c7dd92-76c2-11e6-9b3a-b63f2bd0b775.png)

**Comment by vassvik** [comment]
I'm using LumixEngine's docking extensions, as so many others are, to make an editor-ish thing to be mainly used for my simulations. I wanted to create a minimal project with just the essentials to have a decent looking docking example with as few dependencies as possible, so I removed all the things I didn't need from LumixEngine's docking files and replaced it with simpler things. 

![screenshot_docking](https://cloud.githubusercontent.com/assets/2770871/18421872/e51d64a4-7893-11e6-8f81-99521fb5e465.png)

[Current work-in-progress code](https://github.com/vassvik/imgui_docking_minimal)

**Comment by darkf** [comment]
@thedmd Please release the source, I'm *very* interested in node based editors like this!

**Comment by lapinozz** [comment]
I made a Pixel Art Upscaler using Dear ImGui

https://github.com/lapinozz/ScalaPixel

![demo](https://camo.githubusercontent.com/a4566bbb8a93c2effbed00bfe97963c3e7da6b2d/687474703a2f2f692e696d6775722e636f6d2f6e4d554c6f4c392e676966)

**Comment by ExtasyHosting** [comment]
Still a work in progress...  

![](https://i.gyazo.com/53bb9b2265d2b9b3cc0461355f615c13.png)![](https://i.gyazo.com/a021203d1dd5ef1bc5a4338d529fa61d.gif)

**Comment by bkaradzic** [comment]
@ExtasyHosting Slick! :)

**Comment by BrunoLevy** [comment]
![geobox](https://cloud.githubusercontent.com/assets/19478253/18849484/7a711384-8433-11e6-83e2-75b0d3c5a233.png)

GEOGRAM geometry processing library and utilities.
[[documentation]](http://alice.loria.fr/software/geogram/doc/html/index.html) [[download]](https://gforge.inria.fr/frs/?group_id=5833) [[online demo]](http://webloria.loria.fr/~levy/GEOGRAM/)

**Comment by z350z** [comment]
@thedmd : Similar to Unreal Engine Blueprint  :D

**Comment by thedmd** [comment]
@z350z Yes they are

**Comment by cmaughan** [comment]
![image](https://cloud.githubusercontent.com/assets/120447/19147195/f330b2c0-8bae-11e6-88c6-99d050f87b5a.png)
This is a port of the AntTweakBar rotation widgets.  It probably still needs work, but is self contained inside imgui_orient.cpp/h.  It works by transforming geometry to screen space, so it doesn't need any ImGui API changes.  I had to add a little extra math support (and the original code was flat arrays, so I converted it to use more sensible vector types).  All credit to the original author; this is a really nice rotation widget and 'feels' correct when you spin it; at least to me!
I imagine this will be immediately useful to many folks....
It is on my fork at: https://github.com/cmaughan/imgui, which is mostly the same as the colorpicker branch.
I can imagine this widget would be useful as a model spinner in the main viewport; so an option to remove the background might be nice.  It might also work well as a popup like the color picker.  I've just done the minimum to get it going for now (you can try it in the Test window).

**Comment by ocornut** [comment]
Thanks Chris, very useful !

I am not sure it had been posted but since it is related by Cedric Guillemet also made a 3d-style gizmo to use within game scenes (rather than in UI):

https://github.com/CedricGuillemet/ImGuizmo

![687474703a2f2f692e696d6775722e636f6d2f79346d63566f542e676966](https://cloud.githubusercontent.com/assets/8225057/19147730/ce7813dc-8bb9-11e6-9a06-4958f110673d.gif)

![687474703a2f2f692e696d6775722e636f6d2f6f3871386948712e676966](https://cloud.githubusercontent.com/assets/8225057/19147733/d184a392-8bb9-11e6-93a0-21d439a0b5b9.gif)

**Comment by cmaughan** [comment]
I actually played with Cedric's widgets - the translation and scale ones are very nice.  I found the rotation one confusing though.
I think there's no reason you couldn't use the spin ball in a 3D scene, as just a widget on top of the viewport.  I haven't tried that, but I bet it is quite intuitive.
In terms of integrating with ImGui; I tried to make the source follow similar patterns.  The math bits are reasonably small/clean in the header file (an ImVec3, ImQuat, a few simple math functions).

**Comment by blackhole89** [comment]
I am making a JavaScript-enabled workbench for manipulating graphs ([here](https://github.com/blackhole89/graphicdepictions)):
<img src="https://raw.githubusercontent.com/blackhole89/graphicdepictions/master/screenshots/gd-erdoes-renyi.png" />

The font rendering is done by a modded version of Vuhdo's imgui_freetype branch. The rest of the weird styling is because I previously tried to match the appearance of a clunkier handmade UI library for a different project.

**Comment by luckyycode** [comment]
@edin-p your theme is awesome! could you share color style set please?

**Comment by thatbooisaspy** [comment]
Using imgui for my level editor:
![qyaxz67](https://cloud.githubusercontent.com/assets/2244285/19834658/01eda4ec-9e3f-11e6-9c88-4a4f46bfe010.png)

Inspired by Lumix and uses a modified version of their tabs (with a few bugs fixed)

**Comment by nem0** [comment]
@thatbooisaspy are this bugfixes somwhere to see?

**Comment by thatbooisaspy** [comment]
@nem0 not currently, i havent looked at if you fixed them or not, and i've made quite a few changes to the original source and how the tabs are saved. if i have the time i'll try to find what i fixed and upload it somewhere

**Comment by v71** [comment]
Guys, what do you think if dearimgui supported officialy the opengl vector lib ? ( sorry if this post is not pertinent )

**Comment by ocornut** [comment]
@v71 
> Guys, what do you think if dearimgui supported officialy the opengl vector lib ? ( sorry if this post is not pertinent )
I don't understand the question and I don't know what the opengl vector lib is. Please open a new issue to clarify.

**Comment by williamledoux** [comment]
> I don't understand the question and I don't know what the opengl vector lib is. Please open a new issue to clarify.

Maybe @v71 was talking about  [OpenVG](https://www.khronos.org/openvg/).

**Comment by v71** [comment]
My proposal is to use or to create a lib for imgui to handle vectors , since you talked about quaternions and the possibility to adopt inside the same lib. Since would be more appropriate to consider a plantform agnostic library i proposed the glm library, which is written for opengl coding but it is not dependent by this specific library, it has a syntax nealry equal to the glsl language and supports many common 2d operations. the site is http://glm.g-truc.net/0.9.8/index.html. In case it would be overkill to include i have written simpler 2d and 3d vector libs just in case you were interested into including this kind of computations.,

**Comment by Cthutu** [comment]
This is a bad idea. And shouldn't be in this issue. Imgui is library
agnostic and should remain so

**Comment by cmaughan** [comment]
I personally use glm, and I suspect many here do, but agreed it shouldn't be in the library.
There are simple ways to interchange with glm anyway.  You just have to set the following defines before including imgui:

#define IM_VEC2_CLASS_EXTRA                                                 \
ImVec2(const glm::vec2& f) { x = f.x; y = f.y; }                            \
operator glm::vec2() const { return glm::vec2(x, y); }

#define IM_VEC3_CLASS_EXTRA                                                 \
ImVec3(const glm::vec3& f) { x = f.x; y = f.y; z = f.z;}          \
operator glm::vec3() const { return glm::vec3(x,y,z); }

#define IM_VEC4_CLASS_EXTRA                                                 \
ImVec4(const glm::vec4& f) { x = f.x; y = f.y; z = f.z; w = f.w; }          \
operator glm::vec4() const { return glm::vec4(x,y,z,w); }

#define IM_QUAT_CLASS_EXTRA                                                 \
ImQuat(const glm::quat& f) { x = f.x; y = f.y; z = f.z; w = f.w; }          \
operator glm::quat() const { return glm::quat(w,x,y,z); }

**Comment by ocornut** [comment]
Problem solved (this is a bad idea for five millions reasons). Please do not discuss it in this thread anymore (if you have something to add please move it to a separate thread as already requested).

**Comment by nem0** [comment]
Lumix Engine - Improved curve editor by @JakubLukas
![curves](https://cloud.githubusercontent.com/assets/153526/20189663/80e83e2c-a77e-11e6-9448-35191f05d5d7.gif)

**Comment by Flix01** [comment]
It's still not perfect, (and usable only for small code snippets); however I've finally coded an (almost) functional ImGui::TextInputMultiline(...) clone that supports syntax highlighting.

![imgui_textinputwithsyntaxhighlighting](https://cloud.githubusercontent.com/assets/9608982/20314669/3fd1c274-ab5b-11e6-9156-a519fbf4714f.png)

**Comment by v71** [comment]
Why is it usable for small code snippets ? what prevents it from handling larger texts ?
i think this would be a great add on for imgui

**Comment by ocornut** [comment]
InputText need to be rewritten, perhaps ditching stb_textedit to be efficient for large text. Right now editing a 100k file is unpractical (display somehow works but inputs suffer).

**Comment by nem0** [comment]
Animation graph in Lumix
![animgraph](https://cloud.githubusercontent.com/assets/153526/20415440/83d2450a-ad38-11e6-847b-f3afe70954c2.gif)
![animgraph](https://cloud.githubusercontent.com/assets/153526/20415447/8fb58922-ad38-11e6-94d6-d2ad453cc104.jpg)

**Comment by Flix01** [comment]
Just a slightly modified ImGui::PlotHistogram(...) here:
![imguiplothistogram1](https://cloud.githubusercontent.com/assets/9608982/20539504/8810c7c0-b0f5-11e6-8048-e3d77c617846.png)

**Comment by ocornut** [comment]
Some of the tools used for our game [Wonder Boy: The Dragon's Trap](http://www.TheDragonsTrap.com)

![wb_20161124b](https://cloud.githubusercontent.com/assets/8225057/20628927/33e14cac-b329-11e6-80f6-9524e93b048a.png)

**Comment by v71** [comment]
The windows are a little bit invasive in my opinion , i think we badly need an official docking / tabbed layout handling system, by the way the game graphics looks awesome, congrats to you all guys

**Comment by ocornut** [comment]
@v71 yeah it's not really how we work on a day to day basis, everybody uses 3-4 mains windows (the animator never uses the audio tools, etc.). Actually we probably have 3 times more tools but I'm not allowed to show them all now, just did a selection of some nice stuff here.

After I'm done with the game I will look at the docking/tabbing. I'm pretty confident that the unofficial extensions are doing a good job at this tho :)

**Comment by Cthutu** [comment]
Wonder Boy was a video game character from the 80s.  Did you license it?
Or is this a new Wonder Boy?

**Comment by v71** [comment]
Probably the "wonder boy" phrase has not been patented

**Comment by ocornut** [comment]
We got a licence for the game and trademark from LAT+SEGA.

**Comment by paulsapps** [comment]
@ocornut that is AWESOME :D. Slightly off topic but do you mind if I ask how you handle collisions in your game? I assume you need per pixel accuracy? If so do you use the "render everything with a unique colour and read back a pixel" trick?

**Comment by v71** [comment]
May i ask you why you didn't use Unity, i do not have anything against it, but i really dislike it, do you think it would have been better ?? ( i really doubt it ).

**Comment by ocornut** [comment]
@paulsapps I do not WANT per pixel accuracy for a retro styled games. Bounding boxes are perfectly fines. 
@v71 No specific reason, I like doing thing my own way.

**Comment by ocornut** [comment]
@paulsapps I do not WANT per pixel accuracy for a retro styled games. Bounding boxes are perfectly fines. 
@v71 No specific reason, I like doing thing my own way.

**Comment by ocornut** [comment]
The Harfang 3D engine includes ImGui apparently
https://www.harfang3d.com/doc/0.9.9/man.EngineDebugger.html

Screenshot (I think it is rather old)
![man scene_debugger](https://cloud.githubusercontent.com/assets/8225057/20750600/01db9e6a-b6f8-11e6-9073-03215450542b.jpg)

**Comment by edselmalasig** [comment]
can you make a twitter feed for screenshots that u approve like in khronos.org

**Comment by ocornut** [comment]
From https://twitter.com/hb3p8/status/806572629259943936
![czgfbbywqaavxqm jpg orig](https://cloud.githubusercontent.com/assets/8225057/21047677/7b667d80-be0b-11e6-80c5-80e9055310c8.jpg)

@edselmalasig good idea, may look into that.

**Comment by JWki** [comment]
Nothing too fancy, using imgui for a web based gcode debugging tool for my thesis:

![gcodedebug](https://cloud.githubusercontent.com/assets/8052931/21079781/d0f19252-bf9c-11e6-8462-963dd3b6c5a7.png)

**Comment by edin-p** [comment]
![screen1](https://cloud.githubusercontent.com/assets/12642134/21346800/3c188ee0-c6a6-11e6-980f-05fadcc30b99.jpg)

![screen2](https://cloud.githubusercontent.com/assets/12642134/21346831/6434012a-c6a6-11e6-8ef4-40dcf9bcfabf.jpg)

Working on UI.. Light and Dark theme. Would be nice if I could draw the borders for items in the current BeginChild without inheriting from the owner window but still looks nice :)

**Comment by ocornut** [comment]
Nice! Please post your theme in #707 when you are happy with them.

One of the remaining task for 1.50 is to make BorderSize a regular ImGuiStyle value. I have a stash somewhere but it is leading to some problems with preserving current beginchild/popup expectation and not adding too many entries in the style.

**Comment by ongamex** [comment]
@edin-p Please post the white theme ^_^

**Comment by spacepluk** [comment]
@edin-p that's beautiful :)

**Comment by Cthutu** [comment]
@edin-p Did you ever release your docking code?

**Comment by edin-p** [comment]
I did, it is on my github... But I only worked for about 5 minutes - minor ui adjustments since I put it on github so its quite broken.. It works well enough for prototyping other stuff just/to test so that's what I did - continued writing the rest of the engine

**Comment by Cthutu** [comment]
@edin-p Which font do you use, if you don't mind me asking?

**Comment by edin-p** [comment]
It's OpenSans - Bold

**Comment by jangalomph** [comment]
![2016-12-28_12-28-51](https://cloud.githubusercontent.com/assets/22388398/21529788/56cdc7f6-ccf9-11e6-93cb-2ed98543ace3.png)

Working on an advanced vector field generator (no 3d preview yet). UI still needs plenty of work, but the colors are quite nice i think!

**Comment by v71** [comment]
Aren't you the guy who posted this very image on gamedev ?

**Comment by ongamex** [comment]
@v71 Yep (ImoogiBG on gamedev)

**Comment by BrunoLevy** [comment]
![geocod](https://cloud.githubusercontent.com/assets/19478253/21724574/37b70d44-d435-11e6-9c38-05040bc20040.jpg)

GEOCOD: an environment for teaching programming to kids.
Uses ImGUI+LUA (+Emscripten for the web version)

Try it online: https://members.loria.fr/Bruno.Levy/GEOCOD/
It is part of GEOGRAM: http://alice.loria.fr/software/geogram/doc/html/index.html

**Comment by AndrewBelt** [comment]
This is kind of a meta suggestion, but you can simply link here
https://github.com/ocornut/imgui/issues?utf8=✓&q="Post%20your%20screenshots%20/%20code%20here"
on each screenshot thread so you don't have to update the description each time.

**Comment by ocornut** [comment]
> This is kind of a meta suggestion, but you can simply link here on each screenshot thread so you don't have to update the description each time.

Correct, but the direct links are 1 click away instead of 2 so it is more convenient for the user :)

**Comment by ocornut** [comment]
**This is Part 3 out of 4. I am splitting issues to reduce loading times and locked the old ones.**

Screenshots Part 1 #123
Screenshots Part 2 #539
Screenshots Part 3 #772
Screenshots Part 4 #973

**Comment by yyangdid** [comment]
mark
